web/users.go

```diff
@@ -51,12 +51,7 @@ func (s *AutograderService) getUsers() (*pb.Users, error) {
 
 // getUserByCourse returns the user matching the given GitHub login if
 // the user is enrolled in the given course.
-<<<<<<< HEAD
 func (s *AutograderService) getStudentByCourse(request *pb.CourseUserRequest, currentUser *pb.User) (*pb.User, error) {
```

Comment:
Isn't this better named `getUserByCourse`?

This check:

```go
	if !(currentUser.IsAdmin || s.isTeacher(currentUser.ID, course.ID)) {
		return nil, ErrInvalidUserInfo
	}
```

Is only checking that the teacher or an admin is performing the operation... So technically, this can also return another teacher as well... or?

Reply:
Well, sure. As far as I understand, the teaching staff will be present on the channel, as well as all the students. Am I wrong?

Reply:
Yes, there needs to be a bootstrap I guess... but maybe also TAs can use `!register github_login` to sign up ... which would call this method...